cpp/src/arrow/dataset/file_ipc.h

```diff
@@ -59,6 +61,16 @@ class ARROW_DS_EXPORT IpcFileFormat : public FileFormat {
   std::shared_ptr<FileWriteOptions> DefaultWriteOptions() override;
 };
 
+/// \brief Per-scan options for IPC fragments
+class ARROW_DS_EXPORT IpcFragmentScanOptions : public FragmentScanOptions {
+ public:
+  std::string type_name() const override { return kIpcTypeName; }
+
+  /// Options passed to the IPC file reader.
+  /// included_fields, memory_pool, and use_threads are ignored.
+  std::shared_ptr<ipc::IpcReadOptions> options;
+};
+
```

Comment:
Note I didn't bother exposing this to Python/R since the IPC options in general aren't really exposed.